Parse-Dashboard/Authentication.js

```diff
@@ -90,6 +101,14 @@ function authenticate(userToTest, usernameOnly) {
     //the provided auth matches one of the users
     this.validUsers.find(user => {
       let isAuthenticated = false;
+      if (user.mfa && !usernameOnly) {
+        if (!userToTest.otpCode) {
+          otpMissing = true;
+        }
+        if (!authenticator.verify({ token:userToTest.otpCode, secret: user.mfa })) {
+          otpValid = false;
```

Comment:
@dblythy Really great you kept up with this!

Just reading through (not tested), I believe you made a common logical error.  If there are multi-users with multiple OTP-enabled, the first user will work, but I believe any subsequent user matches will fail.  3 quick fixes come to mind:

1) Put an `else` statement after line 110, with `otpValid = true;`.  (Least change option).

2) Default value otpValid as false on line 95: `let otpValid = false;` and make line 108 condition: `authenticator.verify({ token:userToTest.otpCode, secret: user.mfa })` (remove the `!`).

3) Put an additional condition that matches the username before running through the OTP validation (can be most efficient option).


Also, not super ideal using sh1 for the totp, but that's totally okay with good security practices.

Thanks again for contributing!

Reply:
Nice pickup! My bad, thank you!